src/streaming/net/HTTPLoader.js

```diff
@@ -233,72 +253,40 @@ function HTTPLoader(cfg) {
+                            const initialNumberOfAttempts = mediaPlayerModel.getRetryAttemptsForType(HTTPRequest.MEDIA_SEGMENT_TYPE);
+                            if (initialNumberOfAttempts === remainingAttempts) {
+                                eventBus.trigger(Events.ATTEMPT_BACKGROUND_SYNC);
+                            }
+                        }
+                    } catch (e) {}
+
+                    _retriggerRequest();
                 }
             });
-        };
-
-        /**
-         * Fired when a request has been aborted, for example because the program called XMLHttpRequest.abort().
-         */
-        const _onabort = function () {
-            _oncomplete();
-            _addHttpRequestMetric(httpRequest, httpResponse, traces);
-            if (progressTimeout) {
-                clearTimeout(progressTimeout);
-                progressTimeout = null;
-            }
-            if (config.abort) {
-                config.abort(requestObject);
-            }
-        };
-
-        /**
-         * Fired when progress is terminated due to preset time expiring.
-         * @param event
-         */
-        const _ontimeout = function (event) {
-            _oncomplete();
 
-            let timeoutMessage;
-            // We know how much we already downloaded by looking at the timeout event
-            if (event.lengthComputable) {
-                let percentageComplete = (event.loaded / event.total) * 100;
-                timeoutMessage = 'Request timeout: loaded: ' + event.loaded + ', out of: ' + event.total + ' : ' + percentageComplete.toFixed(3) + '% Completed';
-            } else {
-                timeoutMessage = 'Request timeout: non-computable download size';
-            }
-            logger.warn(timeoutMessage);
-            _addHttpRequestMetric(httpRequest, httpResponse, traces);
-            _retriggerRequest();
         };
 
-        /**
-         * Fired when the request encountered an error.
-         */
-        const _onerror = function () {
-            // If we get a 404 to a media segment we should check the client clock again and perform a UTC sync in the background.
-            try {
-                if (httpResponse.status === 404 && settings.get().streaming.utcSynchronization.enableBackgroundSyncAfterSegmentDownloadError && requestObject.type === HTTPRequest.MEDIA_SEGMENT_TYPE) {
-                    // Only trigger a sync if the loading failed for the first time
-                    const initialNumberOfAttempts = mediaPlayerModel.getRetryAttemptsForType(HTTPRequest.MEDIA_SEGMENT_TYPE);
-                    if (initialNumberOfAttempts === remainingAttempts) {
-                        eventBus.trigger(Events.ATTEMPT_BACKGROUND_SYNC);
-                    }
-                }
-            } catch (e) {}
+        const _updateResourceTimingInfo = function() {
+            requestObject.startDate = requestStartTime;
```

Comment:
I suggest to add these three updates to the `requestObject` to another function as they are not related to the `resourceTiming`. That way we keep the code cleaner in terms of responsibilities,

Reply:
Done